### ISSUE TITLE
Use some smart pointers (unique_ptr) in DropOnTimeline::onExecute() / getNextDocFromImage()

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#4852]
I think we could use some `std::unique_ptr<Doc>` in `DropOnTimeline::onExecute()` to store the `srcDoc` (and remove the `delete srcDoc` + change some functions that receive `**srcDoc` to receive a `std::unique_ptr&` directly/or return it).

Might be a possible a memory leak in case of this [break statement](https://github.com/aseprite/aseprite/blob/9b5920eef3edfd5c020b5e020c8989da5318d71a/src/app/cmd/drop_on_timeline.cpp#L193)? If that is true, a `std::unique_ptr` should fix that.

**Comment by martincapello** [comment]
Indeed, there was a chance of memory leak. Should be fixed once merged  https://github.com/martincapello/aseprite/commit/01ffc3f57c05f594f0e86e724fea62462dbcfb7c
